client.go

```diff
@@ -196,7 +195,23 @@ func (c *Client) post(ctx context.Context, config *apiConfig, apiReq interface{}
 	}
 
 	req.URL.RawQuery = q
-	return ctxhttp.Do(ctx, c.httpClient, req)
+	return c.do(ctx, req)
+}
+
+func (c *Client) do(ctx context.Context, req *http.Request) (*http.Response, error) {
+	client := c.httpClient
+	if client == nil {
+		client = http.DefaultClient
+	}
+	resp, err := client.Do(req.WithContext(ctx))
+	if err != nil {
+		select {
+		case <-ctx.Done():
+			err = ctx.Err()
+		default:
+		}
```

Comment:
No need for this select statement. Just return `err`. If the context is canceled, that error will be returned by the client.

See:
https://play.golang.org/p/5giVHFAmnNq